File: lua/1118.lua
local close = false

while not close do
    local count = 0
    local sum = 0.0
    local choose = 0

    while count < 2 do
        local v = io.read("*n")

        if v < 0 or v > 10 then
            print("nota invalida")
        else
            sum = sum + v
            count = count + 1
        end
    end

    print(string.format("media = %.2f", sum / 2))
    count = 0
    sum = 0

    while true do
        print("novo calculo (1-sim 2-nao)")
        choose = io.read("*n")

        if choose == 2 or choose == 1 then
          if choose == 2 then
            close = true
          end

          break  -- Exit the inner loop
        end
    end
end

File: go/1118.go
package main

import "fmt"

func main() {
  var close bool
  var count, choose int
  var v, sum float64
  for {
    fmt.Scanf("%f", &v)
    if v < 0 || v > 10 {
      fmt.Println("nota invalida")
      continue
    }
    sum += v
    count += 1

    if count != 2 {
      continue
    }

    fmt.Printf("media = %.2f\n", sum / 2)
    count = 0
    sum = 0

    for {
      fmt.Println("novo calculo (1-sim 2-nao)")
      fmt.Scanf("%d", &choose)

      if choose != 2 && choose != 1 {
        continue
      }

      if choose == 2 {
        close = true
      }

      break
    }

    if close {
      break
    }
  }

}
